src/app/fdctl/run/tiles/fd_gossip.c

```diff
@@ -0,0 +1,628 @@
+  fd_pubkey_t   identity_public_key;
+
+  /* Includes Ethernet, IP, UDP headers */
+  ulong gossip_buffer_sz;
+  uchar gossip_buffer[ FD_NET_MTU ];
+
+  ushort net_id;
+  uchar src_mac_addr[6];
+  fd_net_hdrs_t hdr[1];
+
+  fd_keyguard_client_t  keyguard_client[1];
+};
+typedef struct fd_gossip_tile_ctx fd_gossip_tile_ctx_t;
+
+FD_FN_CONST static inline ulong
+scratch_align( void ) {
+  return 128UL;
+}
+
+FD_FN_PURE static inline ulong
+loose_footprint( fd_topo_tile_t const * tile FD_PARAM_UNUSED ) {
+  return 1UL * FD_SHMEM_GIGANTIC_PAGE_SZ;
+}
+
+FD_FN_PURE static inline ulong
+scratch_footprint( fd_topo_tile_t const * tile FD_PARAM_UNUSED ) {
+  ulong l = FD_LAYOUT_INIT;
+  l = FD_LAYOUT_APPEND( l, alignof(fd_gossip_tile_ctx_t), sizeof(fd_gossip_tile_ctx_t) );
+  l = FD_LAYOUT_APPEND( l, fd_gossip_align(), fd_gossip_footprint() );
+  l = FD_LAYOUT_APPEND( l, fd_contact_info_table_align(), fd_contact_info_table_footprint( FD_PEER_KEY_MAX ) );
+  return FD_LAYOUT_FINI( l, scratch_align() );
+}
+
+FD_FN_CONST static inline void *
+mux_ctx( void * scratch ) {
+  return (void*)fd_ulong_align_up( (ulong)scratch, alignof(fd_gossip_tile_ctx_t) );
+}
+
+static void
+send_packet( fd_gossip_tile_ctx_t * ctx,
+             uint                   dst_ip_addr,
+             ushort                 dst_port,
+             uchar const *          payload,
+             ulong                  payload_sz,
+             ulong                  tsorig ) {
+  uchar * packet = fd_chunk_to_laddr( ctx->net_out_mem, ctx->net_out_chunk );
+
+  fd_memcpy( packet, ctx->hdr, sizeof(fd_net_hdrs_t) );
+  fd_net_hdrs_t * hdr = (fd_net_hdrs_t *)packet;
+
+  hdr->udp->net_dport = dst_port;
+
+  memset( hdr->eth->dst, 0U, 6UL );
+  memcpy( hdr->ip4->daddr_c, &dst_ip_addr, 4UL );
+  hdr->ip4->net_id = fd_ushort_bswap( ctx->net_id++ );
+  hdr->ip4->check  = 0U;
+  hdr->ip4->net_tot_len  = fd_ushort_bswap( (ushort)(payload_sz + sizeof(fd_ip4_hdr_t)+sizeof(fd_udp_hdr_t)) );
+  hdr->ip4->check  = fd_ip4_hdr_check( ( fd_ip4_hdr_t const *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4 ) );
+
+  ulong packet_sz = payload_sz + sizeof(fd_net_hdrs_t);
+  fd_memcpy( packet+sizeof(fd_net_hdrs_t), payload, payload_sz );
+  hdr->udp->net_len   = fd_ushort_bswap( (ushort)(payload_sz + sizeof(fd_udp_hdr_t)) );
+  hdr->udp->check = fd_ip4_udp_check( *(uint *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4->saddr_c ), 
+                                      *(uint *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->ip4->daddr_c ), 
+                                      (fd_udp_hdr_t const *)FD_ADDRESS_OF_PACKED_MEMBER( hdr->udp ), 
+                                      packet + sizeof(fd_net_hdrs_t) );
+
+  ulong tspub = fd_frag_meta_ts_comp( fd_tickcount() );
+  ulong sig   = fd_disco_netmux_sig( 0U, 0U, dst_ip_addr, DST_PROTO_OUTGOING, FD_NETMUX_SIG_MIN_HDR_SZ );
+  fd_mcache_publish( ctx->net_out_mcache, ctx->net_out_depth, ctx->net_out_seq, sig, ctx->net_out_chunk, packet_sz, 0UL, tsorig, tspub );
```

Comment:
Use `fd_mux_publish` for primary output link

Reply:
done